algo/src/test/java/com/hartwig/serve/sources/ckb/ActionableTrialFactoryTest.java

```diff
@@ -0,0 +1,128 @@
+                createTrialWithMultipleLocations("Recruiting", "Belgium", "Recruiting", "India", "Suspended");
+        ClinicalTrial terminatedTrial = createTrial("Terminated", "Netherlands", "Suspended");
+        assertEquals(1, ActionableTrialFactory.countriesToInclude(trialPotentiallyOpenInBelgium).size());
+        assertEquals(0, ActionableTrialFactory.countriesToInclude(trialNotOpenInNetherlands).size());
+        assertEquals(0, ActionableTrialFactory.countriesToInclude(trialPotentiallyOpenInUnitedStates).size());
+        assertEquals(0, ActionableTrialFactory.countriesToInclude(trialWithdrawnInUnitedStates).size());
+        assertEquals(0, ActionableTrialFactory.countriesToInclude(terminatedTrial).size());
+        assertEquals(1, ActionableTrialFactory.countriesToInclude(trialPotentiallyOpenInBelgiumAndIndia).size());
+        assertEquals(2, ActionableTrialFactory.countriesToInclude(trialPotentiallyOpenInBelgiumAndGermany).size());
+        assertEquals(1, ActionableTrialFactory.countriesToInclude(trialPotentiallyOpenInBelgiumSuspendedInGermany).size());
+        assertEquals(1, ActionableTrialFactory.countriesToInclude(trialPotentiallyOpenInBelgiumSuspendedInIndia).size());
+    }
+
+    @Test
+    public void canDetermineIfHasVariantRequirementTypeToInclude() {
+        assertTrue(ActionableTrialFactory.hasVariantRequirementTypeToInclude(extractVariantRequirementDetails("required"),
+                createBasicEntry()));
+        assertTrue(ActionableTrialFactory.hasVariantRequirementTypeToInclude(extractVariantRequirementDetails("partial - required"),
+                createBasicEntry()));
+        assertFalse(ActionableTrialFactory.hasVariantRequirementTypeToInclude(extractVariantRequirementDetails("excluded"),
+                createBasicEntry()));
+        assertFalse(ActionableTrialFactory.hasVariantRequirementTypeToInclude(extractVariantRequirementDetails("diagnostic"),
+                createBasicEntry()));
+        assertFalse(ActionableTrialFactory.hasVariantRequirementTypeToInclude(extractVariantRequirementDetails("partial - excluded"),
+                createBasicEntry()));
+        assertFalse(ActionableTrialFactory.hasVariantRequirementTypeToInclude(extractVariantRequirementDetails("allowed"),
+                createBasicEntry()));
+    }
+
+    @NotNull
+    private static CkbEntry createBasicEntry() {
+        return CkbTestFactory.createEntry("BRAF", "BRAF V600E", "BRAF V600E", "sensitive", "Actionable", "AB", "AB", "A", "DOID:162");
+    }
+
+    @NotNull
+    private static List<VariantRequirementDetail> extractVariantRequirementDetails(@NotNull String requirementType) {
+        ClinicalTrial trial = CkbTestFactory.createTrialWithRequirementType(Lists.newArrayList(ImmutableVariantRequirementDetail.builder()
+                .profileId(0)
+                .requirementType(requirementType)
+                .build()));
+        return trial.variantRequirementDetails();
```

Comment:
There are various simplifications you can make here:
 - You first create VariantRequirementDetail in line 99, then create a ClinicalTrial, then return the variant requirement details anymore. The creation of the ClinicalTrial is unnecessary.
 - You have quite some duplicated code in the top level test. You can replace this with a function "hasRequirementTypeToInclude(string)", by expanding the createBasicEntry function